spanner/transaction.go

```diff
@@ -297,6 +309,7 @@ func (t *txReadOnly) ReadWithOptions(ctx context.Context, table string, keys Key
 					DataBoostEnabled:    dataBoostEnabled,
 					DirectedReadOptions: directedReadOptions,
 					OrderBy:             orderBy,
+					LockHint:            lockHint,
```

Comment:
I see that lock hints can only be used with RW transactions (https://github.com/googleapis/google-cloud-go/blob/0c17df27b06c4d67bc9a52ffad88fc58589d0096/spanner/apiv1/spannerpb/spanner.pb.go#L2009).
But `ReadWithOptions` method can be used for both RO and RW transactions.
So what will be the behaviour if this is used for RO transaction? Will it be no-op or will it throw any error?

One alternative is to set this field only when this method is called with in a read write transaction.

Reply:
Backend will throw an error if this is set in RO txn. 

>> One alternative is to set this field only when this method is called with in a read write transaction.

Can you pls send me code pointets for this?

Reply:
`ReadOptions` can be set at client level and request level. 
1. When customer sets `lock hint` in `ReadOptions` at `request level` with in a `RO` txn then they get an error and that is right behaviour.
2. But when they set it at client level, customer expects that this `lockhint` gets set only for allowed txn (i.e `RW`) and gets ignored for `RO`. We did the same in case of directed reads as well. So, one thing we can do is in `client.go` file when the `lockhint` is set (this is client level setting) then that can be overriden to default with in the readonly transaction methods [here](https://github.com/googleapis/google-cloud-go/blob/18fa7e4961d055939078833d0442a415fac96ae6/spanner/client.go#L593).
WDYT?

Reply:
sg, done.